sound/soc/sof/ipc4-topology.c

```diff
@@ -2837,17 +2830,32 @@ static int sof_ipc4_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 		gtw_attr->lp_buffer_alloc = pipeline->lp_mode;
 		fallthrough;
 	case SOF_DAI_INTEL_ALH:
+	{
+		struct sof_ipc4_alh_configuration_blob *blob;
+
+		blob = (struct sof_ipc4_alh_configuration_blob *)ipc4_copier->copier_config;
 		/*
 		 * Do not clear the node ID when this op is invoked with
 		 * SOF_DAI_CONFIG_FLAGS_HW_FREE. It is needed to free the group_ida during
 		 * unprepare.
 		 */
 		if (flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) {
+			int i;
+
+			for (i = 0; i < SOF_IPC4_DMA_DEVICE_MAX_COUNT; i++) {
+				if (!blob->alh_cfg.mapping[i].device) {
+				        blob->alh_cfg.mapping[i].device =
+				        	copier_data->gtw_cfg.node_id;
+					blob->alh_cfg.mapping[i].device |=
+						SOF_IPC4_NODE_INDEX(data->dai_node_id);
```

Comment:
@ranj063 this won't work for LNL+ , see line 1725. https://github.com/thesofproject/linux/commit/54818ce980984f185e7b596dbd5ad3f57d4c1c67#diff-694481cfb73dd1682c248c54855215595dc96ca4ee65b1d3ef9cdc4a478158e0R1725

maybe it's just simpler to save the dai_node_id somewhere and reset the node_id in the unprepare?

Reply:
@plbossart I changed the logic to save the ALH dai index during hw_params instead. Hopefully this is a better solution